modules/account-lib/src/coin/cspr/transaction.ts

```diff
@@ -36,14 +37,28 @@ export class Transaction extends BaseTransaction {
       Uint8Array.from(Buffer.from(keys.prv, 'hex')),
     );
     const signedDeploy = DeployUtil.signDeploy(this._deploy, secpKeys);
-    this.addSignature(signedDeploy.approvals[signedDeploy.approvals.length - 1].signature);
+    this._signatures.push(signedDeploy.approvals[signedDeploy.approvals.length - 1].signature);
   }
 
   /**
-   * Add a signature to this transaction
+   * Add a signature to this transaction and to and its deploy
    * @param {string} signature The signature to add, in string hex format
+   * @param {KeyPair} keyPair The key pair that created the signature
    */
-  addSignature(signature: string): void {
+  addSignature(signature: string, keyPair: KeyPair): void {
+    const pub = keyPair.getKeys().pub;
+    const signatureBuffer = Uint8Array.from(Buffer.from(signature, 'hex'));
+    const pubKeyBuffer = Uint8Array.from(Buffer.from(pub, 'hex'));
+    const parsedPublicKey = Keys.Secp256K1.parsePublicKey(pubKeyBuffer, 'raw');
+    const pubKeyHex = Keys.Secp256K1.accountHex(parsedPublicKey);
+    if (pubKeyHex.slice(2) !== _.toLower(pub)) {
```

Comment:
Hello Luis, i dont remeber which prefix does remove here, it is secpt256k1 doesnt it? could you clarify this?
I think you can use SECP256K1_PREFIX from [account-lib/src/coin/cspr/constants.ts)](https://github.com/BitGo/BitGoJS/blob/STLX-2076-cspr-fix-addSignature/modules/account-lib/src/coin/cspr/constants.ts)

Reply:
Added a new helper function in utils to centralize this logic